core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -656,7 +669,7 @@ class Partition(val topicPartition: TopicPartition,
    *
    * @return true if the high watermark has been updated
    */
-  private def maybeExpandIsr(followerReplica: Replica, followerFetchTimeMs: Long): Boolean = {
+  private def maybeExpandIsr(followerReplica: Replica, followerFetchTimeMs: Long): Unit = {
     inWriteLock(leaderIsrUpdateLock) {
       // check if this replica needs to be added to the ISR
       leaderLogIfLocal match {
```

Comment:
We can change this to a `foreach`

Reply:
We should also update the scaladoc to remove the result type.